### PR TITLE
Various functional test fixes

### PR DESCRIPTION
**Pull request by scottbommarito**
1 - Increased max attempts for `CheckIfPackageVersionExistsInSource`. We are waiting for the search service here and we halved the search service reload frequency from every 5 minutes to every 10 minutes, so some tests began failing.

2 - Added `-noninteractive` to all `nuget.exe` calls. This is necessary to prevent test case stalling, specifically with the `delete` command. Also did some minor restructuring to the class for beautification as requested.

3 - Restructured `FindPackagesByIdTest`. It now bypasses the search service entirely.

4 - Removed an unnecessary call to `GetLatestStableVersion` that was causing issues in `UploadAndDownLoadPackageWithMinClientVersion`

@skofman1 @ryuyu @xavierdecoster @shishirx34

### CODE REVIEW

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs [hunk 1]
```diff
@@ -68,7 +68,7 @@ public bool CheckIfPackageVersionExistsInSource(string packageId, string version
             SemanticVersion semVersion;
             var success = SemanticVersion.TryParse(version, out semVersion);
             const int interval = 30;
```
Comment (ryuyu) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs]: nit: intervalInSeconds

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs [hunk 2]
```diff
@@ -46,7 +47,7 @@ public CommandlineHelper(ITestOutputHelper testOutputHelper)
         {
             WriteLine("Uploading package " + packageFullPath + " to " + sourceName);
 
-            var arguments = string.Join(string.Empty, PushCommandString, @"""" + packageFullPath + @"""", SourceSwitchString, sourceName, ApiKeySwitchString, EnvironmentSettings.TestAccountApiKey);
```
Comment (ryuyu) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: Is there a reason we use a string,Join with an empty separator and spaces in the static strings, and not use a string.Join with space separator and not spaces in the static strings?

File: tests/NuGetGallery.FunctionalTests/ODataFeeds/V2FeedExtendedTests.cs [hunk 3]
```diff
@@ -48,10 +48,16 @@ private bool CanUploadToSite()
...
 
                 TestOutputHelper.WriteLine("Uploaded package '{0}'", packageId);
-                await _clientSdkHelper.UploadNewPackageAndVerify(packageId, "2.0.0");
+                await _clientSdkHelper.UploadNewPackage(packageId, "2.0.0");
+
+                _clientSdkHelper.VerifyPackageExistsInSource(packageId);
+                _clientSdkHelper.VerifyPackageExistsInSource(packageId, "2.0.0");
+
+                // Wait for two minutes to guarantee there's time for the feed to update with the package.
+                Thread.Sleep(2 * 60 * 1000);
```
Comment (shishirx34) [inline tests/NuGetGallery.FunctionalTests/ODataFeeds/V2FeedExtendedTests.cs]: Don't like sleep. Do a poll frequently and fail after 2 minutes.

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs (hunk 2 above)
Comment (scottbommarito) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: i agree, it's very ugly but it's outside the scope of this pr and i don't want to bother with it

File: tests/NuGetGallery.FunctionalTests/Commandline/NuGetCommandLineTests.cs [hunk 4]
```diff
@@ -4,6 +4,7 @@
 using System;
 using System.ComponentModel;
 using System.IO;
+using System.Threading;
```
Comment (skofman1) [inline tests/NuGetGallery.FunctionalTests/Commandline/NuGetCommandLineTests.cs]: what's the reason for this change?

File: tests/NuGetGallery.FunctionalTests.Core/packages.config [hunk 5]
```diff
@@ -2,7 +2,7 @@
 <packages>
   <package id="Microsoft.Web.Xdt" version="2.1.1" targetFramework="net45" />
   <package id="Newtonsoft.Json" version="7.0.1" targetFramework="net45" />
-  <package id="NuGet.Core" version="2.8.6" targetFramework="net45" />
+  <package id="NuGet.Core" version="2.14.0" targetFramework="net45" />
```
Comment (skofman1) [inline tests/NuGetGallery.FunctionalTests.Core/packages.config]: what's the reason for this change?

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs [hunk 6]
```diff
@@ -96,7 +97,7 @@ public CommandlineHelper(ITestOutputHelper testOutputHelper)
         /// <returns></returns>
         public async Task<ProcessResult> UpdateNugetExeAsync()
         {
-            var arguments = string.Join(string.Empty, UpdateCommandString, "-self");
+            var arguments = string.Join(string.Empty, UpdateCommandString, "-self", NonInteractiveSwitchString);
             return await InvokeNugetProcess(arguments);
```
Comment (skofman1) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: Instead of changing all the methods using InvokeNuGetProcess, change InvokeNuGetProcess. It should get a flag nonInteractive which is true by default, and adds this argument. This way the change is only in one place.

File: tests/NuGetGallery.FunctionalTests.Core/packages.config (hunk 5 above)
Comment (scottbommarito) [inline tests/NuGetGallery.FunctionalTests.Core/packages.config]: upgraded nuget core because it was thought that it might have been the catalyst for another issue i'm seeing with these test cases but still haven't fixed

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs (hunk 2 above)
Comment (scottbommarito) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: i have changed it to a much prettier style.

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs [hunk 7]
```diff
@@ -67,24 +67,24 @@ public bool CheckIfPackageVersionExistsInSource(string packageId, string version
...
-            const int maxAttempts = 15;
+            const int intervalSec = 30;
+            const int maxAttempts = 30;
 
             if (success)
             {
                 try
                 {
-                    WriteLine("Starting package verification checks ({0} attempts, interval {1} seconds).", maxAttempts, interval);
+                    WriteLine("Starting package verification checks ({0} attempts, interval {1} seconds).", maxAttempts, intervalSec);
```
Comment (shishirx34) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs]: Can't you just use $"Starting package... {maxAttempts} attempts..." and so on? here and everywhere below too.

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs [hunk 8]
```diff
@@ -73,7 +84,12 @@ public CommandlineHelper(ITestOutputHelper testOutputHelper)
         /// <returns></returns>
         public async Task<ProcessResult> InstallPackageAsync(string packageId, string sourceName)
         {
-            var arguments = string.Join(string.Empty, InstallCommandString, packageId, SourceSwitchString, sourceName);
+            WriteLine("Installing package " + packageId + " from " + sourceName);
```
Comment (shishirx34) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: Same as above.. use {packageId} .. ?

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs [hunk 9]
```diff
@@ -108,16 +151,19 @@ public CommandlineHelper(ITestOutputHelper testOutputHelper)
...
         /// <param name="timeout">Timeout in seconds (default = 6min).</param>
         /// <returns></returns>
-        public async Task<ProcessResult> InvokeNugetProcess(string arguments, string workingDir = null, int timeout = 360)
+        public async Task<ProcessResult> InvokeNugetProcess(List<string> arguments, string workingDir = null, int timeout = 360)
         {
             var nugetProcess = new Process();
             var pathToNugetExe = Path.Combine(Environment.CurrentDirectory, NugetExePath);
 
-            WriteLine("The NuGet.exe command to be executed is: " + pathToNugetExe + " " + arguments);
+            arguments.Add(NonInteractiveSwitchString);
```
Comment (shishirx34) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: not absolutely necessary, but pass this as a flag to this method default to true and then add to argument list, I think thats what @skofman1 suggested in previous iteration.

File: tests/NuGetGallery.FunctionalTests.Core/NuGetGallery.FunctionalTests.Core.csproj [hunk 10]
```diff
@@ -40,8 +40,8 @@
       <HintPath>..\..\packages\Newtonsoft.Json.7.0.1\lib\net45\Newtonsoft.Json.dll</HintPath>
     </Reference>
     <Reference Include="NuGet.Core, Version=2.8.60717.93, Culture=neutral, PublicKeyToken=31bf3856ad364e35, processorArchitecture=MSIL">
-      <SpecificVersion>False</SpecificVersion>
       <HintPath>..\..\packages\NuGet.Core.2.8.6\lib\net40-Client\NuGet.Core.dll</HintPath>
+      <Private>True</Private>
```
Comment (shishirx34) [inline tests/NuGetGallery.FunctionalTests.Core/NuGetGallery.FunctionalTests.Core.csproj]: needed?

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs (hunk 9 above)
Comment (scottbommarito) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/CommandlineHelper.cs]: i can't see a situation in which automated tests would want an interactive cli...it just causes the tests to stall

File: tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs (hunk 7 above)
Comment (scottbommarito) [inline tests/NuGetGallery.FunctionalTests.Core/Helpers/ClientSdkHelper.cs]: Actually, no, because of how `ITestOutputHelper`, an external `xunit` class, is written. (`WriteLine` is a wrapper around `ITestOutputHelper.WriteLine`.) There could potentially be some special logic inside `ITestOutputHelper` that handles this differently than a normal message and I don't want to overwrite it.
